linkerd/addr/src/lib.rs

```diff
@@ -258,3 +258,22 @@ mod tests {
         }
     }
 }
+
+#[cfg(fuzzing)]
+pub mod fuzz_logic {
+    use super::*;
+
+    pub fn fuzz_addr_1(fuzz_data: &str) {
+        if let Ok(addr) = Addr::from_str(fuzz_data) {
+            addr.is_loopback();
+            addr.to_http_authority();
+            addr.is_loopback();
+            addr.socket_addr();
```

Comment:
i'm not very familiar with `cargo fuzz` idioms and practices, so this could be a stupid question, but would it make sense to test each of these methods in a separate fuzz target so that it's clearer where failures occurred?

Reply:
We will get nice stack traces out that shows us where the bug occurs - so it will be easy to debug issues in this way.

In general we don't want to have too small fuzzers as well - especially once we put them up on clusterfuzz (oss-fuzz) which will throw a bit of computing power on it. We want to make sure each of the fuzzers (or those that we enable for OSS-Fuzz) have a fair amount of coverage.